### PR TITLE
Add `insert1_skip_full_duplicates` and remove `recursive_search`

### LINKED ISSUE #42: Generalized paramset insert?
Currently, multiple elements (and sometimes multiple schemas within each) have a paramset insert function
- [ephys](https://github.com/datajoint/element-array-ephys/blob/main/element_array_ephys/ephys_acute.py#L368-L386) 
- [calcium imaging](https://github.com/datajoint/element-calcium-imaging/blob/main/element_calcium_imaging/imaging.py#L96-L117)
- [deeplabcut](https://github.com/datajoint/element-deeplabcut/blob/main/element_deeplabcut/train.py#L147-L197) (under revision)

Should element-interface host a generalized version of this process?

### PR DESCRIPTION
**Pull request by CBroz1**
As discussed in #42. Stopgap pending feature to datajoint-python ([feature request](https://github.com/datajoint/datajoint-python/issues/1049)).

### CODE REVIEW

File: element_interface/utils.py [hunk 1]
```diff
@@ -142,3 +150,26 @@ def recursive_search(key, dictionary):
             if a is not None:
                 return a
     return None
+
+
+def insert1_skip_full_duplicates(table, entry):
+    """
+    Inserts one entry into table, ignoring duplicates only if all other entries match.
```
Comment (dimitri-yatsenko) [inline element_interface/utils.py]: Usually "entries" describes rows.
```suggestion
    Inserts one entry into table, ignoring duplicates only if all the secondary attributes match.
```

File: element_interface/utils.py [hunk 2]
```diff
@@ -134,6 +138,10 @@ def recursive_search(key, dictionary):
         :param dictionary: nested dictionary
         :return a: value of the input argument `key`
     """
+    logger.warn(
```
Comment (dimitri-yatsenko) [inline element_interface/utils.py]: This may be overkill. The function was used in one place, so simply removing should not cause anyone any harm.

**Comment by CBroz1** [comment]
Hi @dimitri-yatsenko - If this looks good to you, I'll add to `element-optogenetics` and plan to incorporate in other elements down the road.

File: element_interface/utils.py [hunk 3]
```diff
@@ -146,25 +147,24 @@ def ingest_csv_to_table(
...
-    """Return value for key in a nested dictionary
-
-    Search through a nested dictionary for a key and returns its value.  If there are
-    more than one key with the same name at different depths, the algorithm returns the
-    value of the least nested key.
-
-    Args:
-        key (str): Key used to search through a nested dictionary
-        dictionary (dict): Nested dictionary
+def insert1_skip_full_duplicates(table, entry):
```
Comment (dimitri-yatsenko) [inline element_interface/utils.py:150]: Awesome. This will be integrated into DataJoint shortly. https://github.com/datajoint/datajoint-python/issues/1049

**Review by dimitri-yatsenko — APPROVED** [review]

**Comment by CBroz1** [comment]
Hi @dimitri-yatsenko - Is this ready to merge?

**Comment by CBroz1** [comment]
We tried this out in practice and ran into issues when the secondary attributes were duplicated with a novel primary key.
The desired functionality would skip entry if the secondary attributes already existed. Tabling this for now.

**Pull request closed without merging**